ui/hooks/src/use-logged-in.ts

```diff
@@ -10,8 +12,50 @@ import { useGetLogin } from './use-login.new';
  */
 export function useLoggedIn() {
   const loginQuery = useGetLogin();
-  const data = useMemo(() => {
-    return loginQuery.data;
-  }, [loginQuery.data]);
-  return { isLoggedIn: !!data?.id, loggedInProfileId: (data?.id || '').toLowerCase() };
+  const queryClient = useQueryClient();
+
+  const myProfileQuery = useGetMyProfileQuery(null, {
+    enabled: !!loginQuery.data?.id,
+    //users who hasnt filled in their account info has no akashaProfile(null), but they are considered logged in if the viewer is not null
+    select: response => response.viewer,
+  });
+
+  console.log('useLoggedIn', 'useGetLogin', loginQuery, myProfileQuery);
+
+  useEffect(() => {
+    if (loginQuery.data?.id && !myProfileQuery.data) {
+      console.log('first condition true');
+
+      const refetchProfileData = async () => {
+        const { data } = await myProfileQuery.refetch();
+        return data;
+      };
+
+      if (!refetchProfileData()) {
+        queryClient.setQueryData([LOGIN_STATE_KEY], null);
+      }
+    }
+
+    if (!loginQuery.data?.id && myProfileQuery.data) {
+      console.log('second condition true');
+
+      const resetQueryPromise = queryClient.resetQueries(useGetMyProfileQuery.getKey());
+      const timeoutPromise = new Promise((resolve, _) => {
+        setTimeout(resolve, 3000);
+      });
+
+      const res = async () => Promise.race([resetQueryPromise, timeoutPromise]);
+      res();
+    }
```

Comment:
:D TBH I am not sure what exactly is going on. I thought the race condition is between resetting useGetMyProfileQuery query and the logout mutation ? 

And btw on line 47 I don't see the use of async as you aren't awaiting anything.